special-pages/pages/new-tab/app/favorites/components/Favorites.js

```diff
@@ -265,10 +271,20 @@ function useVisibleRows(rows, rowHeight, safeAreaRef) {
         }
         const startIndex = Math.floor(start / rowHeight);
         const endIndex = Math.min(Math.ceil(end / rowHeight), rows.length);
-        setVisibleRange({ start: startIndex, end: endIndex });
+
+        // don't set state if the offset didn't change
+        setVisibleRange((prev) => {
+            if (startIndex !== prev.start || endIndex !== prev.end) {
+                return { start: startIndex, end: endIndex };
+            }
+            return prev;
+        });
     }
 
     useLayoutEffect(() => {
+        // ignore scrolling events if collapsed
+        if (expansion === 'collapsed') return;
+
```

Comment:
I realised we were fire scroll events even when the widget is in it's 'collapsed' state